### PR TITLE
Feature/skale 3418 proxy on schain

### PR DESCRIPTION
**Pull request by DimaStebaev**
Add upgradeable proxies.

### CODE REVIEW

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by yavrsky — DISMISSED** [review]

File: proxy/predeployed/src/ima_predeployed/contracts/token_manager_linker.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def _setup(self, deployer_address: str, linker_addres: str) -> None:
+        self._write_uint256(self.INITIALIZED_SLOT, 1)
+        self._setup_role(self.ROLES_SLOT, self.DEFAULT_ADMIN_ROLE, [deployer_address])
+        self._setup_role(self.ROLES_SLOT, self.REGISTRAR_ROLE, [deployer_address])
+        self._write_address(self.MESSAGE_PROXY_SLOT, MESSAGE_PROXY_FOR_SCHAIN_ADDRESS)
+        self._write_address(self.LINKER_ADDRESS_SLOT, linker_addres)
+        self._write_addresses_array(
+            self.TOKEN_MANAGERS_SLOT,
+            [TOKEN_MANAGER_ETH_ADDRESS, TOKEN_MANAGER_ERC20_ADDRESS, TOKEN_MANAGER_ERC721_ADDRESS])
```
Comment (payvint) [inline proxy/predeployed/src/ima_predeployed/contracts/token_manager_linker.py]: Looks like no CommunityLocker and Token_manager_ERC1155

File: proxy/generate_precompiled_config.js [hunk 2]
```diff
@@ -231,10 +231,10 @@ const g_arrContracts = [
         address: "0x57ad607c6e90df7d7f158985c3e436007a15d744",
         referenceVariableName: "TokenManagerLinker"
     }, {
-        fileName: "EthERC20.json",
-        filePath: "artifacts/contracts/schain/tokens/EthERC20.sol/EthERC20.json",
+        fileName: "EthErc20.json",
```
Comment (payvint) [inline proxy/generate_precompiled_config.js]: Do we still need generate precompiled config? If we need this - addresses are incorrect - if not - let's remove

File: proxy/contracts/schain/TokenManagers/TokenManagerERC20.sol [hunk 3]
```diff
@@ -140,7 +128,7 @@ contract TokenManagerERC20 is TokenManager {
 
         contractOnSchain.burn(amount);
```
Comment (payvint) [inline proxy/contracts/schain/TokenManagers/TokenManagerERC20.sol]: Maybe we can make _exit function as you added in ERC721/Eth?

**Review by payvint — COMMENTED** [review]
Reviewed - discussed

File: proxy/predeployed/src/ima_predeployed/contracts/token_manager_linker.py (hunk 1 above)
Comment (DimaStebaev) [reply proxy/predeployed/src/ima_predeployed/contracts/token_manager_linker.py]: Added Token_manager_ERC1155 but not CommunityLocker.

File: proxy/generate_precompiled_config.js (hunk 2 above)
Comment (DimaStebaev) [reply proxy/generate_precompiled_config.js]: Yes. It's duplication. Have removed the file.

File: proxy/contracts/schain/TokenManagers/TokenManagerERC20.sol (hunk 3 above)
Comment (DimaStebaev) [reply proxy/contracts/schain/TokenManagers/TokenManagerERC20.sol]: Changed.

**Review by payvint — APPROVED** [review]
No comments... Bellissimo!!!

**Review by sergiy-skalelabs — APPROVED** [review]

**Pull request merged**